aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -1545,44 +1545,99 @@ namespace {
     }
   }
 
-  void check_scale(const Tensor& mat, const Tensor& scale, const int dim, const int arg_idx, const int scale_multiplier=1) {
-    if (mat.dim() == 2) {
+  void check_input_and_scale_size(const Tensor& mat_a, const Tensor& mat_b, const Tensor& scale_a, const Tensor& scale_b, const std::optional<at::Tensor>& offs) {
```

Comment:
https://github.com/pytorch/pytorch/blob/1b88da1cac30dec473cbdca4d9efb9b117cb8cdb/aten/src/ATen/native/cuda/Blas.cpp#L1086

I think we should do something equivalent to this, we will need to support MX dtypes for groupd_mm on b100(not in this pr) but ultimately we will have something pretty similar for scaled grouped mm